starter-code/src/functions-and-arrays.js

```diff
@@ -1,100 +1,465 @@
 // Find the maximum
+var nb1 = 2;
+var nb2 = 3;
+
+function maxOfTwoNumbers(nb1, nb2) {
+  return Math.max(nb1, nb2);
+}
 
 // Finding Longest Word
 var words = [
-  'mystery',
-  'brother',
-  'aviator',
-  'crocodile',
-  'pearl',
-  'orchard',
-  'crackpot'
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
 ];
 
+function findLongestWord(words) {
+  for (var i = 0; i < words.length; i++) {
+    var long = "";
+    if (words[i].length > long.length) {
+      long = words[i];
+    } else {
+      long = long;
+    }
+  }
+  return long1;
+}
+
+findLongestWord(words);
+
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(numbers) {
+  var sum = 0;
+  for (var i = 0; i < numbers.length; i++) {
+    sum = sum + numbers[i];
+  }
+  return sum;
+}
+
 // Calculate the Average
 
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(numbersAvg) {
+  var sum = 0;
+  if (numbersAvg.length > 0) {
+    for (var i = 0; i < numbersAvg.length; i++) {
+      sum = sum + numbersAvg[i];
+    }
+    var average = sum / numbersAvg.length;
+    return average;
+  }
+}
+
 // Array of Strings
 var wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
 ];
 
+function averageWordLength(wordsArr) {
+  var sum = 0;
+  if (wordsArr.length > 0) {
+    for (var i = 0; i < wordsArr.length; i++) {
+      sum = sum + wordsArr[i].length;
+    }
+    var average = sum / wordsArr.length;
+    return average;
+  }
+}
+
 // Unique Arrays
 var wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring"
 ];
 
+function uniquifyArray(wordsUnique) {
+  if (wordsUnique.length === 0) {
+    return undefined;
+  } else {
+    for (var i = 0; i < wordsUnique.length; i++) {
+      var indices = [];
+      var element = wordsUnique[i];
+      var position = wordsUnique.indexOf(element);
+      while (position != -1) {
+        indices.push(position);
+        position = wordsUnique.indexOf(element, position + 1);
+      }
+      console.log(position);
+      console.log(indices);
+
+      if (indices.length >= 2) {
+        var indicebis = indices[1];
+        wordsUnique.splice(indicebis, 1);
+      }
+    }
+  }
+  return wordsUnique;
+}
+
```

Comment:
It's a pretty tricky way to do it but it's good! 

You could have done it, in a simpler way like this:
```
function uniquifyArray(array) {
  var newArray = [];
  if (array.length === 0) {
    return undefined;
  }
  for (var i = 0; i < array.length; i++) {
    if (newArray.indexOf(array[i]) === -1) {
      newArray.push(array[i]);
    }
  }
  return newArray;
}
```